gensim/test/test_word2vec.py

```diff
@@ -257,6 +257,13 @@ def models_equal(self, model, model2):
         self.assertTrue(numpy.allclose(model[most_common_word], model2[most_common_word]))
 #endclass TestWord2VecModel
 
+    def test_sentences_should_not_be_a_generator(self):
+        """
+        Is sentences a generator object?
+        """
+        gen = (s for s in sentences)
+        self.assertRaises(TypeError, word2vec.Word2Vec(gen))
```

Comment:
This won't work -- by the time `assertRaises` gets its arguments, `Word2Vec(gen)` has already been called and failed. the generator must be an extra argument.